npc/re/other/kachua_key.txt

```diff
@@ -0,0 +1,71 @@
+//===== rAthena Script =======================================
+//= Dynamic NPC: Kachua's Secret Box
+//===== Description: =========================================
+//- [Official conversion]
+//= Kachua's Secret Box is a Gachapon NPC.
+//= It lets the player exchange a Kachua's Secret Key for a random item.
+//= It also gives a Kachua's Mileage Coupon for each pull.
+//===== Changelogs: ==========================================
+//= 1.0 First version. [secretdataz]
+//============================================================
+
+sec_in02,126,178,3	script	Kachua's Secret Box#bm	4_TREASURE_BOX,{
+	mes "A Secret Box where no one knows what's inside.";
+	mes "^4d4dffYou can open this box by consuming <ITEM>[Kachua's Secret Key]<INFO>23919</INFO></ITEM>.^000000";
+	next;
+	switch(select("^4d4dffOpen the box 1 time (1 Kachua Secret Key)^000000","^4d4dffOpen the box 10 times (10 Kachua Secret Keys)^000000")) {
+		case 1:
+			mes "^FF0000[Notice]^000000";
+			mes "^FF0000Exchange one random item with one Kachua's Secret Key.^000000";
+			mes "^FF0000The item exchanged above cannot be withdrawn, nor can it be exchanged to Kachua's Secret Key.^000000";
+			next;
+			if(select("Keep going.","Stop conversation.") == 2) {
+				mes "You have decided not to open the Box.";
+				close;
+			} else if (countitem("K_Secret_Key") < 1) {
+				mes "Not enough <ITEM>[Kachua Secret Key]<INFO>23919</INFO></ITEM>.";
+				close;
+			} else {
+				if (checkweight(1201,1) == 0 || ((MaxWeight - Weight) * 100 / MaxWeight) < 10) {
+					mes "^4d4dffPlease make sure you have enough space in your inventory.^000000";
+					close;
+				}
+				delitem("K_Secret_Key", 1);
+				getgroupitem(IG_KACHUA_SECRET_BOX);
```

Comment:
```suggestion
				consumeitem(.@box$);
```

Reply:
consumeitem can't be used with getitem, getgroupitem (too many run_script, backup lost)
 https://github.com/rathena/rathena/pull/6840